arbitrum-docs/arbos/l1-to-l2-messaging.mdx

```diff
@@ -1,64 +1,160 @@
+# L1 To L2 Messaging
 
-A retryable is submitted via the `Inbox`'s `createRetryableTicket` method. In the common case, a Retryable's submission is followed by an attempt to execute the transaction (i.e., an _"auto-redeem"_). If the attempt fails or isn't scheduled after the Retryable is submitted, anyone can try to _redeem_ it, by calling the [`redeem`](./precompiles#ArbRetryableTx) method of the [`ArbRetryableTx`](./precompiles#ArbRetryableTx) precompile. The party requesting the redeem provides the gas that will be used to execute the Retryable. If execution of the Retryable succeeds, the Retryable is deleted. If execution fails, the Retryable continues to exist and further attempts can be made to redeem it. If a fixed period (currently one week) elapses without a successful redeem, the Retryable expires and will be [automatically discarded][discard_link], unless some party has paid a fee to [_renew_][renew_link] the Retryable for another full period. A Retryable can live indefinitely as long as it is renewed each time before it expires.
+## Retryable Tickets
+
+Retryable tickets are Arbitrum's canonical method for creating L1 to L2 messages, i.e., L1 transactions that initiate a message to be executed on L2. A retryable can be submitted for a fixed cost (dependent only on its calldata size) paid at L1; its _submission_ on L1 is separable / asynchronous with its _execution_ on L2. Retryables provide atomicity between the cross chain operations; if the L1 transaction to request submission succeeds (i.e. does not revert) then the execution of the Retryable on L2 has a strong guarantee to ultimately succeed as well.
+
+## Retryable Tickets Lifecycle
+
+Here we walk through the different stages of the lifecycle of a retryable ticket; (1) submission, (2) auto-redemption, and (3) manual redemption.
+
+
+### Submission
+
+1. Creating a retryable ticket is initiated with a call (direct or internal) to the `createRetryableTicket` function of the `inbox` contract. A ticket is garanteed to be created if this call succeeds. Here, we describe parameters that need to be carefully set.
+Note that, this function forces the sender to provide a _reasonable_ amount of funds (at least enough to submitting, and _attempting_ to executing the ticket), but that doesn't guarantee a successful auto-redemption.
+
+    - `deposit (l1CallValue)`: Not a real function parameter, it is rather the callValue that is sent along with the transaction
```

Comment:
It might be only me, but I find this a bit confusing, specially by being the first one in the list. I was expecting to see parameters of the function and the first one is described as "not being a function parameter". Also, calling it "deposit" makes it hard for me to realize it's the call value sent with the tx. I would have maybe called it "l1 callvalue (also referred to as deposit)" or something like that, because there are many references to "deposit" in the rest of the document.

Again, maybe it's just me, so disregard the comment if it seems clear enough for others !

Reply:
The reason for l1CallValue being the first in the list is it's used in other params definitions, e.g., l2CallValue.
So I guess it still makes sense to keep it there. But I'll def change add ` (also referred to as deposit)`. We also refer to this as deposit in the SDK code so I think that makes sense.